src/platform_impl/macos/view.rs

```diff
@@ -433,6 +460,36 @@ extern "C" fn set_marked_text(
             marked_text.initWithString(string);
         };
         *marked_text_ref = marked_text;
+
+        // Update ViewState with new marked text
+        let state_ptr: *mut c_void = *this.get_ivar("winitState");
+        let state = &mut *(state_ptr as *mut ViewState);
+        let preedit_string = id_to_string(string);
+        let (cursor_position, text_offset) = (
+            selected_range.location as usize,
+            selected_range.length as usize,
+        );
+        state.ime_state = Some(ImeState {
+            text: preedit_string.clone(),
+            cursor_start: cursor_position,
+            cursor_end: cursor_position + text_offset,
+        });
+        if state.ime_state.is_none() {
```

Comment:
I can't be `None`, since you just assigned `Some` to it. I'm not sure what you're doing here.

Reply:
This was my mistake.